modules/fscloud/variables.tf

```diff
@@ -297,11 +299,16 @@ variable "target_service_details" {
   }
   validation {
     condition = alltrue([
-      for target_service_name, attributes in var.target_service_details :
-      contains(["cloud-object-storage", "codeengine", "container-registry", "containers-kubernetes", "containers-kubernetes-cluster", "containers-kubernetes-management", "databases-for-cassandra", "databases-for-elasticsearch", "databases-for-enterprisedb", "databases-for-etcd", "databases-for-mongodb", "databases-for-mysql", "databases-for-postgresql", "databases-for-redis", "event-notifications", "hs-crypto", "iam-identity", "is", "logdna", "logdnaat", "messagehub", "messages-for-rabbitmq", "mqcloud", "secrets-manager", "sysdig-monitor", "sysdig-secure"], target_service_name) if attributes.region != null
+      for target_service_name, attributes in var.target_service_details : (
+        contains(["cloud-object-storage", "codeengine", "container-registry", "containers-kubernetes", "containers-kubernetes-cluster", "containers-kubernetes-management", "databases-for-cassandra", "databases-for-elasticsearch", "databases-for-enterprisedb", "databases-for-etcd", "databases-for-mongodb", "databases-for-mysql", "databases-for-postgresql", "databases-for-redis", "event-notifications", "hs-crypto", "iam-identity", "is", "logdna", "logdnaat", "messagehub", "messages-for-rabbitmq", "mqcloud", "secrets-manager", "sysdig-monitor", "sysdig-secure"], target_service_name) &&
+        (
+          !(attributes.region != null && attributes.geography != null)
+        )
+      )
     ])
-    error_message = "Provide a valid target service name that supports region attribute."
+    error_message = "Both `region` and `geography` cannot be set simultaneously for the target service."
```

Comment:
Geography is only applicable for `container-registry` and not for any other service. So validation has to be set specifically for `container-registry`.

```
Tgt. Services REGION SUPPORTED: 23
["cloud-object-storage", "codeengine", "container-registry", "containers-kubernetes", "databases-for-elasticsearch", "databases-for-enterprisedb", "databases-for-etcd", "databases-for-mongodb", "databases-for-mysql", "databases-for-postgresql", "databases-for-redis", "event-notifications", "hs-crypto", "iam-identity", "is", "logdna", "logdnaat", "messagehub", "messages-for-rabbitmq", "mqcloud", "secrets-manager", "sysdig-monitor", "sysdig-secure"]

Tgt. Services REGION NOT SUPPORTED: 13
["IAM", "apprapp", "compliance", "context-based-restrictions", "directlink", "dns-svcs", "globalcatalog-collection", "iam-access-management", "iam-groups", "kms", "schematics", "transit", "user-management"]

Tgt. Services GEOGRAPHY SUPPORTED: 1
["container-registry"]
```